homeassistant/components/elmax/common.py

```diff
@@ -0,0 +1,229 @@
+"""Elmax integration common classes and utilities."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from datetime import timedelta
+import logging
+from logging import Logger
+from typing import Any
+
+import async_timeout
+from elmax_api.exceptions import (
+    ElmaxApiError,
+    ElmaxBadLoginError,
+    ElmaxBadPinError,
+    ElmaxNetworkError,
+)
+from elmax_api.http import Elmax
+from elmax_api.model.endpoint import DeviceEndpoint
+from elmax_api.model.panel import PanelEntry, PanelStatus
+
+from homeassistant.components.elmax.const import DEFAULT_TIMEOUT, DOMAIN
+from homeassistant.exceptions import ConfigEntryAuthFailed, HomeAssistantError
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ElmaxCoordinator(DataUpdateCoordinator):
+    """Coordinator helper to handle Elmax API polling."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        logger: Logger,
+        username: str,
+        password: str,
+        panel_id: str,
+        panel_pin: str,
+        name: str,
+        update_interval: timedelta,
+    ) -> None:
+        """Instantiate the object."""
+        self._client = Elmax(username=username, password=password)
+        self._panel_id = panel_id
+        self._panel_pin = panel_pin
+        self._panel_entry = None
+        self._state_by_endpoint = None
+        super().__init__(
+            hass=hass, logger=logger, name=name, update_interval=update_interval
+        )
+
+    @property
+    def panel_entry(self) -> PanelEntry | None:
+        """Return the panel entry."""
+        return self._panel_entry
```

Comment:
This a proxy method to make a protected variable public? Why not make the variable public instead of in that case?
Is there any place where public access is needed? (Could not find any).